Trivial updates

mkykadir opened the pull request:
See each commit for details

Commit:
Trivial: Update test genesis data comment

Commit:
Trivial: Remove `Ledger.addEnrollment`

Geod24 commented:
Can you add the motivation for the second commit ? I'm wondering whether `addEnrollment` belongs to the basic `Ledger`, as a mirror to `addTransaction`. Another approach would be to have a basic `Ledger` without any pool (if that makes sense), and a derived class with pools (given that for a client tracking the chain, a pool might not make sense).

mkykadir commented:
`addEnrollment` of Ledger is only used through `TestLedger` in tests. Source code uses enrollment manager only.

Commit:
Trivial: Remove unnecessary iteration while fee clearing

Genesis block doesn't have fees and block fees are always cleared
with last height included.

linked0 reviewed (approved):
LGTM

omerfirmak commented:
> `addEnrollment` of Ledger is only used through `TestLedger` in tests. Source code uses enrollment manager only.

`TestLedger` already exposes enrollment manager thru the `enrollment_manager` getter. We might as well remove addEnrollment entirely.

omerfirmak commented on source/agora/consensus/Ledger.d:
```diff
@@ -1401,27 +1401,6 @@ public class Ledger
-
-    ***************************************************************************/
-
-    public bool addEnrollment (in Enrollment enroll, in PublicKey pubkey,
```
Commit message should be updated as well, feel free to self merge after that.
